nncf/tensorflow/quantization/algorithm.py

```diff
@@ -218,7 +173,7 @@ def _load_state_without_name(self, state: Dict[str, Any]):
         :param state: Output of `get_state()` method.
         """
         quantizer_setup_state = state[self._state_names.QUANTIZER_SETUP]
-        self._quantizer_setup = QuantizationSetup.from_state(quantizer_setup_state)
+        self._quantizer_setup = SingleConfigQuantizerSetup.from_state(quantizer_setup_state)
```

Comment:
>@vshampor, @ljaljushkin Will this lead to an update of the checkpoints?

Yes, here's discussion in another PR
https://github.com/openvinotoolkit/nncf/pull/836#discussion_r666896902

>SingleConfigQuantizerSetup does not include half_range and narrow_range options. This leads to the fact that without additional information there is no way to restore the state of the quantizer from a quantizer setup. This contradicts the idea of storing quantization setup for loading from them. Please extend SingleConfigQuantizerSetup that it fully describe quantizer state.

It's not just adding it to Setup, but also refactoring the code with config. There's 58312 task for it on @aleksei-kashapov

Reply:
I decided to restore the TF-specific quantization setup. Since it stores `TFQuantizerSpec`s rather than `QuantizerConfig`s, it contains the necessary `half_range`/`narrow_range` info.